packages/flutter/test/semantics/semantics_update_test.dart

```diff
@@ -0,0 +1,227 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  SemanticsUpdateTestBinding();
+
+  testWidgets('Semantics update does not send update for merged nodes.', (WidgetTester tester) async {
+    final SemanticsHandle handle = tester.ensureSemantics();
+    await tester.pumpWidget(
+      MaterialApp(
+        home: MergeSemantics(
+          child: Semantics(
+            label: 'outer',
+            textField: true,
+            // This merge semantics node should not be part of the semantics
+            // update because it is under another merge semantics.
+            child: MergeSemantics(
+              child: Semantics(
+                label: 'inner',
+                textField: true,
+                child: const Text('text'),
+              ),
+            ),
+          ),
+        ),
+      ),
+      null,
+      // Stops right after the warm up frame.
+      EnginePhase.build
+    );
+    // The warm up frame will send update for an empty semantics tree. We
+    // ignore this one time update.
+    SemanticsUpdateBuilderSpy.observations.clear();
+    // Builds the real semantics tree.
+    await tester.pump();
+
+    expect(SemanticsUpdateBuilderSpy.observations.length, 5);
+
+    expect(SemanticsUpdateBuilderSpy.observations.containsKey(0), isTrue);
+    expect(SemanticsUpdateBuilderSpy.observations[0]!.childrenInTraversalOrder.length, 1);
+    expect(SemanticsUpdateBuilderSpy.observations[0]!.childrenInTraversalOrder[0], 1);
+
+    expect(SemanticsUpdateBuilderSpy.observations.containsKey(1), isTrue);
+    expect(SemanticsUpdateBuilderSpy.observations[1]!.childrenInTraversalOrder.length, 1);
+    expect(SemanticsUpdateBuilderSpy.observations[1]!.childrenInTraversalOrder[0], 2);
+
+    expect(SemanticsUpdateBuilderSpy.observations.containsKey(2), isTrue);
+    expect(SemanticsUpdateBuilderSpy.observations[2]!.childrenInTraversalOrder.length, 1);
+    expect(SemanticsUpdateBuilderSpy.observations[2]!.childrenInTraversalOrder[0], 3);
+
+    expect(SemanticsUpdateBuilderSpy.observations.containsKey(3), isTrue);
+    expect(SemanticsUpdateBuilderSpy.observations[3]!.childrenInTraversalOrder.length, 1);
+    expect(SemanticsUpdateBuilderSpy.observations[3]!.childrenInTraversalOrder[0], 4);
```

Comment:
I am surprised that it sends this many semantics nodes. I would have expected 2: the root node and the merged node. Is that because of the `MaterialApp` used in the test? Can we remove it (and replace it with the individual widget we need, e.g. a Directionality widget) to make this test more targeted? Otherwise we will have to update it whenever the semantics of MaterialApp change...